README.md
# NAME

Syntax::Collection::Basic - (deprecated)

<div>
    <p>
    <img src="https://img.shields.io/badge/perl-5.18+-blue.svg" alt="Requires Perl 5.18+" />
    <a href="https://travis-ci.org/Csson/syntax-collection-basic"><img src="https://api.travis-ci.org/Csson/syntax-collection-basic.svg?branch=master" alt="Travis status" /></a>
    <a href="http://cpants.cpanauthors.org/release/CSSON/Syntax-Collection-Basic-0.0601"><img src="http://badgedepot.code301.com/badge/kwalitee/CSSON/Syntax-Collection-Basic/0.0601" alt="Distribution kwalitee" /></a>
    <a href="http://matrix.cpantesters.org/?dist=Syntax-Collection-Basic%200.0601"><img src="http://badgedepot.code301.com/badge/cpantesters/Syntax-Collection-Basic/0.0601" alt="CPAN Testers result" /></a>
    </p>
</div>

# VERSION

Version 0.0601, released 2017-10-31.

# STATUS

Deprecated.

# SYNOPSIS

    use Syntax::Collection::Basic;

Is really

    use strict;
    use warnings;
    use Modern::Perl '2014';
    use true;

# SOURCE

[https://github.com/Csson/syntax-collection-basic](https://github.com/Csson/syntax-collection-basic)

# HOMEPAGE

[https://metacpan.org/release/Syntax-Collection-Basic](https://metacpan.org/release/Syntax-Collection-Basic)

# AUTHOR

Erik Carlsson <info@code301.com>

# COPYRIGHT AND LICENSE

This software is copyright (c) 2017 by Erik Carlsson.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Syntax-Collection-Basic

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Stats & Dependency Changes]

 [Testing]

0.0601 2017-10-31T18:25:30Z UTC
 - Deprecated.

 [Stats & Dependency Changes]
 - 4 files changed, 73 insertions(+), 27 deletions(-)
 - (run req) ~ MooseX::StrictConstructor (any) --> 0.19
 - (run req) ~ perl 5.018 --> 5.018000
 - (test req) + ExtUtils::MakeMaker (any)
 - (test req) + File::Spec (any)
 - (test req) + IO::Handle (any)
 - (test req) + IPC::Open3 (any)
 - (test rec) + CPAN::Meta 2.120900
 - (build req) - Test::Pod
 - (conf req) + ExtUtils::MakeMaker (any)
 - (conf req) - Module::Build::Tiny
 - (dev req) + Badge::Depot 0.0103
 - (dev req) + Badge::Depot::Plugin::Coverage (any)
 - (dev req) + Badge::Depot::Plugin::Cpantesters (any)
 - (dev req) + Badge::Depot::Plugin::Kwalitee (any)
 - (dev req) + Badge::Depot::Plugin::Perl (any)
 - (dev req) + Badge::Depot::Plugin::Travis (any)
 - (dev req) + Dist::Zilla 6.010
 - (dev req) + Dist::Zilla::Plugin::Authority 1.009
 - (dev req) + Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.008
 - (dev req) + Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0200
 - (dev req) + Dist::Zilla::Plugin::CheckChangesHasContent 0.008
 - (dev req) + Dist::Zilla::Plugin::Clean 0.07
 - (dev req) + Dist::Zilla::Plugin::ConfirmRelease (any)
 - (dev req) + Dist::Zilla::Plugin::CopyFilesFromBuild 0.170880
 - (dev req) + Dist::Zilla::Plugin::Deprecated (any)
 - (dev req) + Dist::Zilla::Plugin::DistIller::MetaGeneratedBy (any)
 - (dev req) + Dist::Zilla::Plugin::ExecDir (any)
 - (dev req) + Dist::Zilla::Plugin::Git 2.041
 - (dev req) + Dist::Zilla::Plugin::Git::Check (any)
 - (dev req) + Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.014
 - (dev req) + Dist::Zilla::Plugin::Git::Commit (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Contributors 0.030
 - (dev req) + Dist::Zilla::Plugin::Git::GatherDir (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Push (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Tag (any)
 - (dev req) + Dist::Zilla::Plugin::GithubMeta 0.54
 - (dev req) + Dist::Zilla::Plugin::InstallRelease 0.008
 - (dev req) + Dist::Zilla::Plugin::License (any)
 - (dev req) + Dist::Zilla::Plugin::MakeMaker (any)
 - (dev req) + Dist::Zilla::Plugin::Manifest (any)
 - (dev req) + Dist::Zilla::Plugin::ManifestSkip (any)
 - (dev req) + Dist::Zilla::Plugin::MetaConfig (any)
 - (dev req) + Dist::Zilla::Plugin::MetaJSON (any)
 - (dev req) + Dist::Zilla::Plugin::MetaNoIndex (any)
 - (dev req) + Dist::Zilla::Plugin::MetaProvides::Package 2.004003
 - (dev req) + Dist::Zilla::Plugin::MetaYAML (any)
 - (dev req) + Dist::Zilla::Plugin::NextRelease::Grouped 0.0200
 - (dev req) + Dist::Zilla::Plugin::PodSyntaxTests (any)
 - (dev req) + Dist::Zilla::Plugin::PodWeaver 4.006
 - (dev req) + Dist::Zilla::Plugin::Prereqs (any)
 - (dev req) + Dist::Zilla::Plugin::PromptIfStale 0.054
 - (dev req) + Dist::Zilla::Plugin::Readme (any)
 - (dev req) + Dist::Zilla::Plugin::ReadmeAnyFromPod 0.163250
 - (dev req) + Dist::Zilla::Plugin::RewriteVersion::Transitional 0.007
 - (dev req) + Dist::Zilla::Plugin::RunExtraTests 0.028
 - (dev req) + Dist::Zilla::Plugin::ShareDir (any)
 - (dev req) + Dist::Zilla::Plugin::Test::Compile 2.057
 - (dev req) + Dist::Zilla::Plugin::Test::EOF 0.0501
 - (dev req) + Dist::Zilla::Plugin::Test::EOL 0.18
 - (dev req) + Dist::Zilla::Plugin::Test::NoTabs 0.15
 - (dev req) + Dist::Zilla::Plugin::Test::ReportPrereqs 0.027
 - (dev req) + Dist::Zilla::Plugin::Test::Version 1.05
 - (dev req) + Dist::Zilla::Plugin::TestRelease (any)
 - (dev req) + Dist::Zilla::Plugin::UploadToCPAN (any)
 - (dev req) + Pod::Elemental::Transformer::List (any)
 - (dev req) + Pod::Weaver::Plugin::SingleEncoding (any)
 - (dev req) + Pod::Weaver::Plugin::Transformer (any)
 - (dev req) + Pod::Weaver::PluginBundle::CorePrep (any)
 - (dev req) + Pod::Weaver::Section::Authors (any)
 - (dev req) + Pod::Weaver::Section::Badges 0.0401
 - (dev req) + Pod::Weaver::Section::Collect (any)
 - (dev req) + Pod::Weaver::Section::GenerateSection 1.01
 - (dev req) + Pod::Weaver::Section::Generic (any)
 - (dev req) + Pod::Weaver::Section::Leftovers (any)
 - (dev req) + Pod::Weaver::Section::Legal (any)
 - (dev req) + Pod::Weaver::Section::Name (any)
 - (dev req) + Pod::Weaver::Section::Region (any)
 - (dev req) + Pod::Weaver::Section::Version (any)
 - (dev req) + Test::EOF (any)
 - (dev req) + Test::EOL (any)
 - (dev req) + Test::More 0.88
 - (dev req) + Test::NoTabs 1.4
 - (dev req) + Test::Version 1
 - (dev req) + Test::Warnings 0.026
 - (dev sug) + Dist::Iller 0.1408
 - (dev sug) + Dist::Iller::Config::Author::CSSON 0.0319

0.06 2014-09-18T21:04:34Z CEST
 - More missing prereqs...

0.05 2014-09-02T23:58:19Z CEST
 - Fix missing dependencies

0.04 2014-09-01T18:52:13Z CEST
 - Documentation and dependencies

0.02 2014-08-16T13:13:23Z CEST
 - Fix: List dependencies

0.01 2014-05-29T21:37:03Z CEST
 - Initial release


lib/Syntax/Collection/Moose.pm
use 5.018;

# AUTHORITY

package Syntax::Collection::Moose {
our $VERSION = '0.0602';
    use Syntax::Collection::Basic;

    # see http://blogs.perl.org/users/ovid/2013/09/building-your-own-moose.html
    use Moose();
    use MooseX::StrictConstructor();
    use Moops();
    use Kavorka();
    use Moose::Exporter;

    Moose::Exporter->setup_import_methods(
        with_meta => ['has'],
        also      => ['Moose'],
    );

    sub init_meta {
        my $class = shift;
        my %params = @_;
        my $for_class = $params{'for_class'};

        Moose->init_meta(@_);
        MooseX::StrictConstructor->import({ into => $for_class });
        Moops->import(into => $for_class);
        Kavorka->import({ into => $for_class });
    }

    sub has {
        my $meta = shift;
        my $name = shift;
        my %options = @_;

        $options{'is'} //= 'ro';

        foreach (ref $name eq 'ARRAY' ? @$name : $name) {
            $meta->add_attribute($_, %options);
        }
    }
}

=pod

=head1 STATUS

Deprecated.


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.010.
use strict;
use warnings;

use 5.018000;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "(deprecated)",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Syntax-Collection-Basic",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.018000",
  "NAME" => "Syntax::Collection::Basic",
  "PREREQ_PM" => {
    "Kavorka" => 0,
    "Modern::Perl" => "1.20140107",
    "Moops" => "0.025",
    "Moose" => "2.00",
    "MooseX::StrictConstructor" => "0.19",
    "Syntax::Collector" => "0.005",
    "true" => "0.18"
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "0.0602",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Kavorka" => 0,
  "Modern::Perl" => "1.20140107",
  "Moops" => "0.025",
  "Moose" => "2.00",
  "MooseX::StrictConstructor" => "0.19",
  "Syntax::Collector" => "0.005",
  "Test::More" => "0.96",
  "true" => "0.18"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
